core/src/main/java/org/elasticsearch/tasks/Task.java

```diff
@@ -132,4 +135,16 @@ public Status getStatus() {
     }
 
     public interface Status extends ToXContent, NamedWriteable {}
+
+    public TaskResult result(DiscoveryNode node, Throwable error) throws IOException {
+        return new TaskResult(taskInfo(node, false), error);
+    }
+
+    public TaskResult result(DiscoveryNode node, ActionResponse response) throws IOException {
+        if (response instanceof ToXContent) {
+            return new TaskResult(taskInfo(node, false), (ToXContent) response);
+        } else {
+            throw new IllegalStateException("response has to implement ToXContent for persistence");
```

Comment:
Cannot log here because I don't have access to the logger on the task level, but I will fix [logging in the caller](https://github.com/elastic/elasticsearch/pull/17928/files#diff-3c6e6a9409cf4f8d1ce5637eda1b755aR190) to log this exception properly.